integration_tests/src/main/python/data_gen.py

```diff
@@ -193,6 +200,8 @@ def with_special_pattern(self, pattern, flags=0, charset=sre_yield.CHARSET, weig
         return self.with_special_case(lambda rand : strs[rand.randint(0, length-1)], weight=weight)
 
     def start(self, rand):
```

Comment:
Any requirement to the generated strings according to the specified collation ?

Reply:
IIUC, no. From AI:
In Java, string collation refers to the process of comparing and ordering strings based on specific rules, such as locale and character set. 
In Java, the Collator class does not directly deal with character sets (charsets) like UTF-8 or ISO-8859

I.e.
- Compare two string with ignoring the Upper/Lower case
- Compare two string with trim the leading/tailing spaces

IIUC, Collator only specifies some requirements when doing operations.